### PR TITLE
Tidy page about Poseidon/Firmament scheduler

### PR DESCRIPTION
**Pull request by sftim**
Tidy [Poseidon/Firmament scheduler](https://kubernetes.io/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler/) [[preview](https://deploy-preview-18626--kubernetes-io-master-staging.netlify.com/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler/)]

Also see https://github.com/kubernetes-sigs/poseidon

### CODE REVIEW

**Comment by sftim** [comment]
@anvithks, could you review these changes?

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-18626--kubernetes-io-master-staging.netlify.com/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler/

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 1]
```diff
@@ -15,37 +15,37 @@ The Poseidon/Firmament scheduler is an alternate scheduler that can be deployed
...
-- Poseidon-Firmament Scheduler outperforms the Kubernetes default scheduler by a wide margin when it comes to throughput performance numbers for scenarios where compute resource requirements are somewhat uniform across jobs (Replicasets/Deployments/Jobs). Poseidon-Firmament scheduler end-to-end throughput performance numbers, including bind time, consistently get better as the number of nodes in a cluster increase. For example, for a 2,700 node cluster (shown in the graphs [here](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/benchmark/README.md)), Poseidon-Firmament scheduler achieves a 7X or greater end-to-end throughput than the Kubernetes default scheduler, which includes bind time.     
+### Flow graph scheduling based Poseidon-Firmament scheduler provides the following key advantages:
+- Workloads (pods) are bulk scheduled to enable scheduling at massive scale..
+- Based on the extensive performance test results, Poseidon-Firmament scales much better than the Kubernetes default scheduler as the number of nodes increase in a cluster. This is due to the fact that Poseidon-Firmament is able to amortize more and more work across workloads.
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Line 29, Is there a reason to include `..` at end of sentence?

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 2]
```diff
@@ -58,16 +58,16 @@ As mentioned earlier, Poseidon-Firmament scheduler enables an extremely high thr
 
 Although, Poseidon-Firmament scheduler is capable of scheduling various types of workloads, such as service, batch, etc., the following are a few use cases where it excels the most:
 
-1. For “Big Data/AI” jobs consisting of large number of tasks, throughput benefits are tremendous.  
-2. Service or batch jobs where workload resource requirements are uniform across jobs (Replicasets/Deployments/Jobs).  
+1. For “Big Data/AI” jobs consisting of large number of tasks, throughput benefits are tremendous.
+2. Service or batch jobs where workload resource requirements are uniform across jobs (Replicasets/Deployments/Jobs).
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Could you clean up `ReplicaSets/Deployments/Jobs` (don't use `/` to separate terms)?
nit: ReplicaSet?

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 3]
```diff
@@ -58,16 +58,16 @@ As mentioned earlier, Poseidon-Firmament scheduler enables an extremely high thr
...
 
-## Current Project Stage	
+## Current Project Stage
 
-- We are **not aware of any production deployment** of Poseidon-Firmament scheduler at this time.  
-- Poseidon-Firmament is supported from Kubernetes release 1.6 and works with all subsequent releases.   
+- We are **not aware of any production deployment** of Poseidon-Firmament scheduler at this time.
+- Poseidon-Firmament is supported from Kubernetes release 1.6 and works with all subsequent releases.
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Throughout the page, the scheduler is referred to as `Poseidon-Firmament` not `Poseidon/Firmament`. Are the names interchangeable?

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 4]
```diff
@@ -58,16 +58,16 @@ As mentioned earlier, Poseidon-Firmament scheduler enables an extremely high thr
...
 - Release process for Poseidon and Firmament repos are in lock step.  The current Poseidon release can be found [here](https://github.com/kubernetes-sigs/poseidon/releases) and the corresponding Firmament release can be found [here](https://github.com/Huawei-PaaS/firmament/releases).
 
-## Features Comparison Matrix	
+## Feature comparison matrix
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: style guide, replace `here` with something more meaningful.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 5]
```diff
@@ -67,6 +67,10 @@ Although, Poseidon-Firmament scheduler is capable of scheduling various types of
 - Poseidon-Firmament is supported from Kubernetes release 1.6 and works with all subsequent releases.
 - Release process for Poseidon and Firmament repos are in lock step.  The current Poseidon release can be found [here](https://github.com/kubernetes-sigs/poseidon/releases) and the corresponding Firmament release can be found [here](https://github.com/Huawei-PaaS/firmament/releases).
 
+{{< caution >}}
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Line 68. Should this page link to a third party:
https://github.com/Huawei-PaaS/firmament/releases
The Poseidon repo contains links to the Firmament page.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 3 above)
Comment (anvithks) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: @deepak-vij @dhilipkumars Please check this.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 6]
```diff
@@ -92,26 +94,27 @@ Although, Poseidon-Firmament scheduler is capable of scheduling various types of
...
-## Installation	
-
-For in-cluster installation of Poseidon, please start at the [Installation instructions](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/install/README.md).
-
+## Installation
 
-## Development	
+[Installation](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/install/README.md#Installation) explains how to deploy Poseidon/Firmament to your cluster.
 
-For developers, please refer to the [Developer Setup instructions](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/devel/README.md).
```
Comment (anvithks) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: @sftim Any particular reason for removing this?

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 7]
```diff
@@ -1,75 +1,77 @@
 ---
-title: Poseidon-Firmament - An alternate scheduler
+title: Poseidon/Firmament scheduler
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Sure. I'll go with whatever is preferred.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 6 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: I've added:
```markdown
* See [Poseidon/Firmament](https://github.com/kubernetes-sigs/poseidon#readme) on GitHub for more information.
```
which links to the developer information.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 3 above)
Comment (deepak-vij) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Yeah, that should be fine.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 8]
```diff
@@ -1,75 +1,77 @@
...
+### Flow graph scheduling based Poseidon-Firmament scheduler provides the following key advantages:
+- Workloads (pods) are bulk scheduled to enable scheduling at massive scale..
+- Based on the extensive performance test results, Poseidon-Firmament scales much better than the Kubernetes default scheduler as the number of nodes increase in a cluster. This is due to the fact that Poseidon-Firmament is able to amortize more and more work across workloads.
+- Poseidon-Firmament Scheduler outperforms the Kubernetes default scheduler by a wide margin when it comes to throughput performance numbers for scenarios where compute resource requirements are somewhat uniform across jobs (Replicasets/Deployments/Jobs). Poseidon-Firmament scheduler end-to-end throughput performance numbers, including bind time, consistently get better as the number of nodes in a cluster increase. For example, for a 2,700 node cluster (shown in the graphs [here](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/benchmark/README.md)), Poseidon-Firmament scheduler achieves a 7X or greater end-to-end throughput than the Kubernetes default scheduler, which includes bind time.
```
[the remark this answers is not in the document]
Comment (deepak-vij) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Let us stick to Poseidon-Firmament as the naming convention then. Thanks.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 3 above)
Comment (deepak-vij) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Yes, names are interchangeable. Although, let us stick to Poseidon-Firmament naming convention for all the documentation going forward.

Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Tidied

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Tidied

**Comment by sftim** [comment]
How's this?

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 9]
```diff
@@ -92,26 +94,27 @@ Although, Poseidon-Firmament scheduler is capable of scheduling various types of
...
+2. There is limited opportunities for amortizing work across pods because they are considered for placement individually.
 
 These downsides of pod-by-pod schedulers are addressed by batching or bulk scheduling in Poseidon-Firmament scheduler. Processing several pods in a batch allows the scheduler to jointly consider their placement, and thus to find the best trade-off for the whole batch instead of one pod. At the same time it amortizes work across pods resulting in much higher throughput.
 
-{{< note >}}
-   Please refer to the [latest benchmark results](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/benchmark/README.md) for detailed throughput performance comparison test results between Poseidon-Firmament scheduler and the Kubernetes default scheduler.
-{{< /note >}}
+{{% /capture %}}
+{{% capture whatsnext %}}
+* See [Poseidon/Firmament](https://github.com/kubernetes-sigs/poseidon#readme) on GitHub for more information.
```
Comment (anvithks) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Poseidon/Firmament -> Poseidon-Firmament.  
As per the discussion above, and after input from @deepak-vij.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 7 above)
Comment (anvithks) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Poseidon/Firmament -> Poseidon-Firmament. 
As per input from @deepak-vij, unless there is a style guide that enforces the changes being made.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 9 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Fixed

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 7 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Fixed

**Comment by kbhawkey** [comment]
Hi @sftim , I quickly read through the page again.
If you are tidying up, would you check that all of the headings use sentence case?
There are several places where `Firmament` is misspelled (Current project stage and What's next sections). Good to fix.

**Comment by sftim** [comment]
@kbhawkey I'm hoping this gets it over the line

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 10]
```diff
@@ -23,9 +23,7 @@ Firmament models the scheduling problem as a constraint-based optimization over
 
 Poseidon-Firmament scheduler runs alongside the default Kubernetes scheduler as an alternate scheduler. You can run multiple schedulers run simultaneously.
 
-## Key Advantages
-
-### Flow graph scheduling based Poseidon-Firmament scheduler provides the following key advantages:
+Flow graph scheduling based Poseidon-Firmament scheduler provides the following key advantages:
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Hi @sftim . I read the page. Nearly ready. Take a look.
This change needs a new line at line 27 to render the list 😸 
Suggest rewording line 24 to something like:You can simultaneously run multiple schedulers.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 11]
```diff
@@ -23,14 +23,12 @@ Firmament models the scheduling problem as a constraint-based optimization over
...
+- Poseidon-Firmament Scheduler outperforms the Kubernetes default scheduler by a wide margin when it comes to throughput performance for scenarios where compute resource requirements are somewhat uniform across your workload (Deployments, ReplicaSets, Jobs).
+  Poseidon-Firmament scheduler end-to-end throughput performance, including bind time, gets better as the number of nodes in a cluster increase. For example, for a 2,700 node cluster (shown in [Latest Benchmarking Results](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/benchmark/README.md#latest-benchmarking-results), the Poseidon-Firmament scheduler achieved a 7× or greater end-to-end throughput than the Kubernetes default scheduler, which includes bind time.
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: @sftim , Would be good to fix the last few nits and then push these changes in.
This latest change has a couple of problems:
Line 31: Is this supposed to be an item in the list?
Line 30: The Poseidon-Firmament scheduler outperforms the ... 
Line 31: The link in the second sentence is not set up correctly.
Line 31: @anvithks , Does this make sense?
The Poseidon-Firmament's end-to-end throughput performance and bind time improves as the number of nodes in a cluster increase.

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 12]
```diff
@@ -73,24 +73,25 @@ Poseidon-Firmament scheduler does not provide support for high availability; its
 
 ## Feature comparison {#feature-comparison-matrix}
 
+{{< table caption="Features that are or are not supported by the Kubernetes default scheduler, compared to the Poseidon-Firmament scheduler." >}}
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Looking for a succinct table caption:
Feature comparison of Kubernetes and Poseidon-Firmament schedulers

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 13]
```diff
@@ -73,24 +73,25 @@ Poseidon-Firmament scheduler does not provide support for high availability; its
...
-|Colocation Interference Avoidance|N|N**|Planned in Poseidon-Firmament.|
-|Priority Pre-emption|Y|N**|Partially exists in Poseidon-Firmament versus extensive support in Kubernetes default scheduler.|
-|Inherent Re-Scheduling|N|Y**|Poseidon-Firmament scheduler supports workload re-scheduling. In each scheduling run it considers all the pods, including running pods, and as a result can migrate or evict pods – a globally optimal scheduling environment.|
+|Baseline Scheduling capability in accordance to available compute resources (CPU & Memory) on a node|Y|Y†|**†** Not all Predicates & Priorities are supported with Poseidon-Firmament.|
+|Extreme Throughput at scale|Y†|Y|**†** Bulk scheduling approach scales or increases workload placement. Firmament scheduler offers high throughput when resource requirements (CPU/Memory) for incoming Pods are uniform across ReplicaSets/Deployments/Jobs.|
+|Colocation Interference Avoidance|N|N||
+|Priority Pre-emption|Y|N†|**†** Partially exists in Poseidon-Firmament versus extensive support in Kubernetes default scheduler.|
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: line 85: check: s/pre-emption/preemption
line 86: check: s/re-scheduling/rescheduling
line 91: spelling: s/Poseidon-Firmamemtn/xxxxxxx

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md [hunk 14]
```diff
@@ -91,27 +87,27 @@ Although, Poseidon-Firmament scheduler is capable of scheduling various types of
 |Support for Max-Pod per node|Y|Y|Poseidon-Firmament scheduler seamlessly co-exists with Kubernetes default scheduler.|
 |Support for Ephemeral Storage, in addition to CPU/Memory|Y|Y||
 
+## Installation
 
-## Installation	
-
-For in-cluster installation of Poseidon, please start at the [Installation instructions](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/install/README.md).
+[Installation](https://github.com/kubernetes-sigs/poseidon/blob/master/docs/install/README.md#Installation) explains how to deploy Poseidon-Firmament to your cluster.
```
Comment (kbhawkey) [inline content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: nit: Line 92, This change makes this sentence is vague, `Installation explains how to `. What installation?
The [Poseidon-Firmament installation guide](...) explains how to deploy ...

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 11 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: :ok_hand:

File: content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md (hunk 13 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/poseidon-firmament-alternate-scheduler.md]: Think these are dealt with.

**Comment by kbhawkey** [comment]
@sftim , Looks good. I won't mention any nits 😃 .
/lgtm

**Comment by sftim** [comment]
Thanks @kbhawkey !
